compiler/compile/ResolvedMethod.hpp

```diff
@@ -224,6 +224,7 @@ class TR_ResolvedMethod
 
    virtual TR_ResolvedMethod *getResolvedStaticMethod (TR::Compilation *, int32_t cpIndex, bool * unresolvedInCP = 0);
    virtual TR_ResolvedMethod *getResolvedSpecialMethod(TR::Compilation *, int32_t cpIndex, bool * unresolvedInCP = 0);
+   virtual TR_OpaqueMethodBlock *getTargetMethodFromMemberName(uintptr_t * invokeCacheArray, uintptr_t & appendixObject, bool * isInvokeCacheAppendixNull = 0);
```

Comment:
I don't think this needs to be in OMR. All of the call sites are in OpenJ9, where every `TR_ResolvedMethod` will be a `TR_ResolvedJ9Method` anyway, so call sites with only `TR_ResolvedMethod*` can downcast

Reply:
I originally didn't have this OMR change. However, when I tried doing 
```
         TR_OpaqueMethodBlock *targetMethod =
            reinterpret_cast<TR_ResolvedJ9Method *>(resolvedCaller)->getTargetMethodFromMemberName(
               invokeCacheArray, appendixObject);
```
in J9, I kept getting crashes because somehow we'd end up calling the wrong method (it looked like it was looking up the wrong method in the VFT).

Reply:
You need to use `static_cast` instead of `reinterpret_cast`. The object's address differs from its base subobject because of multiple inheritance

Reply:
Heh probably because of
```
class TR_ResolvedJ9Method : public TR_J9Method, public TR_ResolvedJ9MethodBase
```

Reply:
I updated the J9 code to use `static_cast` so I've removed this change from OMR.